### ISSUE TITLE
tensor'name will be renamed when output layer with activation 

### ISSUE THREAD

**Issue by karterotte** [keras-team/keras#13236]
**System information**  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  linux
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.4
- Keras version:  2.0.8
- Python version:  2.7

I have trained a model using keras and want to use it's pb model file to predict online.
The pb model must be called using tensor's name.
But the output tensor name will be renamed with it's layer's name + "/" + activation name.

training code like:
`model.add(keras.layers.Dense(output_size, activation='softmax', name="output"))`

predicting code like:
```
Session.Runner r = s.runner();
Tensor result = r.fetch("output").run().get(0);
```

**I want the tensor'name will be a fixed name "output"  but it is "output/Softmax"**

I tried to rename layer's name or tensor'name ,but all failed.
How can I name the final output from the model?
Thanks!

**Comment by karterotte** [comment]
For my question:
It can be fixed like this:

```
def freeze_session(session, keep_var_names=None, output_names=None, clear_devices=True):
    graph = session.graph
    with graph.as_default():
        freeze_var_names = list(
            set(v.op.name for v in tf.global_variables()).difference(keep_var_names or []))
        output_names = output_names or []
        if len(output_names) > 0:
            for i in range(output_names):
                # here to rename "output/Softmax" to "output"
                tf.identity(model.model.outputs[i], name=output_names[i])
        output_names += [v.op.name for v in tf.global_variables()]
        input_graph_def = graph.as_graph_def()
        if clear_devices:
            for node in input_graph_def.node:
                node.device = ""
        frozen_graph = graph_util.convert_variables_to_constants(session, input_graph_def,
                                                                 output_names, freeze_var_names)
        return frozen_graph

```
then you can do this to save your model to pb:
```
sess = K.get_session()
constant_graph = freeze_session(sess, output_names=["output"])
with tf.gfile.GFile(model_path, mode='wb') as f:
     f.write(constant_graph.SerializeToString())
```
